pennylane/qchem/vibrational/vscf.py

```diff
@@ -0,0 +1,442 @@
+        for alpha in range(3):
+            d2_rot[alpha, ::] = _rotate_two_body(
+                dipole_data[1][alpha, ::], nmodes, mode_rots, modals=modals
+            )
+        dip_data = [d1_rot, d2_rot]
+
+    if n > 2:
+        d3_rot = np.zeros((3, nmodes, nmodes, nmodes, imax, imax, imax, imax, imax, imax))
+        for alpha in range(3):
+            d3_rot[alpha, ::] = _rotate_three_body(
+                dipole_data[2][alpha, ::], nmodes, mode_rots, modals=modals
+            )
+        dip_data = [d1_rot, d2_rot, d3_rot]
+
+    return dip_data
+
+
+def _rotate_hamiltonian(ham_data, mode_rots, modals):
+    r"""Generates VSCF rotated Hamiltonian.
+
+    Args:
+        ham_data (list[TensorLike[float]]): A list of n-mode expansion of Hamiltonian integrals.
+        mode_rots (list[TensorLike[float]]): list of rotation matrices for all vibrational modes
+        modals (list[int]): A list containing the maximum number of modals to consider for each mode.
+
+    Returns:
+        tuple(TensorLike[float]): a tuple of rotated Hamiltonian integrals
+
+    """
+
+    n = len(ham_data)
+    nmodes = np.shape(ham_data[0])[0]
+
+    h1_rot = _rotate_one_body(ham_data[0], nmodes, mode_rots, modals)
+    h_data = [h1_rot]
+
+    if n > 1:
+        h2_rot = _rotate_two_body(ham_data[1], nmodes, mode_rots, modals)
+        h_data = [h1_rot, h2_rot]
+
+    if n > 2:
+        h3_rot = _rotate_three_body(ham_data[2], nmodes, mode_rots, modals)
+        h_data = [h1_rot, h2_rot, h3_rot]
+
+    return h_data
+
+
+def vscf_hamiltonian(ham_data, dipole_data=None, modals=None, cutoff=None, cutoff_ratio=1e-6):
+    r"""Generates VSCF rotated integrals for vibrational Hamiltonian.
+
+    Args:
+        ham_data (list[TensorLike[float]]): A list of n-mode expansion of Hamiltonian integrals.
+        dipole_data (list[TensorLike[float]]): A list of n-mode expansion of dipole integrals. Default is ``None``.
+        modals (list[int]): A list containing the maximum number of modals to consider for each mode. Defalt to maximum number of modals.
+        cutoff (float): threshold value for including matrix elements into operator.
+        cutoff_ratio (float): Default ratio for zeroing elements which are smaller than this ratio with respect to biggest element in all integrals. Default value is ``1e-6``.
+
+    Returns:
+        List[TensorLike[float]]: List of n-mode expansion of Hamiltonian integrals in VSCF basis.
+
+    """
+
+    if len(ham_data) > 3:
+        raise ValueError(
+            f"Building n-mode Hamiltonian is not implemented for n equal to {len(ham_data)}."
+        )
+
+    if dipole_data is not None:
+        if len(dipole_data) > 3:
+            raise ValueError(
+                f"Building n-mode dipole is not implemented for n equal to {len(dipole_data)}."
+            )
+
+    nmodes = np.shape(ham_data[0])[0]
+
+    imax = np.shape(ham_data[0])[1]
+    max_modals = nmodes * [imax]
+    if modals is None:
+        modals = max_modals
```

Comment:
You can simply cover these branches by using `@pytest.mark.parametrize`.

Reply:
The problem was that using max_modals takes a long time for calculation, and thus the test to run. I generated some data for a small number of modals, and will add a test for that.